src/nti/app/pyramid_zope/i18n/__init__.py

```diff
@@ -0,0 +1,68 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+"""
+Support for application-level (request, context and user based) internationalization.
+
+Negotiation
+===========
+
+The process of finding a locale to use is somewhat complex. It is all
+driven around the interface
+:class:`zope.i18n.interfaces.IUserPreferredLanguages` and its mutable
+subclass
+:class:`zope.i18n.interfaces.IModifiableUserPreferredLanguages`. These
+are combined with a :class:`zope.i18n.interfaces.INegotiator`, given a
+context and a list of available languages, to determine the best
+matching locale to use by taking the intersection of the preferred
+languages of the context with the available languages (individual
+translation utilities handle fallback and default languages).
+
+Zope provides an implementation of preferred languages for (Zope)
+requests in
+:class:`zope.publisher.browser.ModifiableBrowserLanguages`. This uses
+the HTTP ``Accept-Language`` header to determine a language. We let
+Pyramid requests also have this implementation through our
+compatibility shims in :mod:`nti.pyramid_zope.i18n`.
+
+However, there are cases where we may not want to rely on the browser
+to have the right setting, either for testing, or to support broken
+browsers, or to explicitly enable user preferences. Supporting user
+preferences is easy, an implementation of mutable preferences is
+provided for :class:`.IUser` objects. For temporary testing or for the
```

Comment:
Is the `IUser` bit still accurate?

Reply:
No. And I can't find where it ever would have been 😿  (We'd need an implementation of `IUserPreferredLanguages` or `IUserModifiablePreferredLanguages` adapting the IUser — now the IPrincipal.)